### DISCUSSION TITLE
RE engine audio hooking(??) modding

### DISCUSSION THREAD

**Discussion by demondji** [community/community#135414] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I would really like to mod the re-engine so I can implement a vocoder to a character's voice lines. I already have a vocoder code to use and I'm a really inexperienced coder/ modder so if anyone could help me or point me in the direction of a framework I can use that would fantastic.

**Comment by geniuszly** [comment]
Modding the RE Engine, especially for something like audio hooking, can be challenging, but it's definitely possible with the right tools and knowledge. 
#### Understanding RE Engine Modding
- **RE Tool**: A community tool that helps in modding games running on the RE Engine (e.g., Resident Evil 2 Remake, Resident Evil 3 Remake). It allows you to unpack and repack game files, including audio files.
- **Fluffy Mod Manager**: Another tool that helps in managing mods for RE Engine games. It's mainly for texture and model mods but can be helpful in organizing your work.
#### Audio Hooking and Processing
- **FMOD Studio**: RE Engine uses FMOD for audio, so understanding how FMOD works can be crucial. You'll need to unpack the game's audio files, modify them with your vocoder, and repack them.
- **Custom Scripting**: Since you already have vocoder code, you’ll need to write scripts to apply this vocoder to the extracted voice lines. This could involve converting the extracted files into a format your vocoder can handle, processing them, and converting them back.
#### Modding Community Resources 
- **Modding Forums**: Places like the Resident Evil Modding community forums are excellent for finding specific tools, tutorials, and help from experienced modders.
- **Discord Communities**: Many RE Engine modders hang out in dedicated Discord servers. Joining one can give you access to a lot of community knowledge and support.
#### Development Setup
- Make sure you have a good environment set up for development. You'll need tools for unpacking/repacking game files, a good text editor or IDE for scripting, and potentially audio editing software.
#### Notes
Given that you’re new to coding and modding, start small. Experiment with simple mods first, like replacing sound files or tweaking settings, before moving on to more complex tasks like implementing your vocoder.

**Reply by demondji** [reply]
thank you very much for this reply genuinely helps a lot, quick question, lets say for the custom script for the vocoder, can I have the task of converting the in-game files to something the vocoder can understand an automated task?

**Reply by geniuszly** [reply]
You can automate the task of converting in-game audio files to a format that the vocoder can understand within a custom script for RE Engine modding, you can try the following: 
1. Use a tool or script to extract the in-game audio files from the RE Engine's archive files. The RE Engine typically stores assets in specific file formats, such as `.pak` or `.bin`. Tools like RE Toolset or custom Python scripts can help with extraction.
2. After extraction, you can automate the conversion of these audio files into a format suitable for the vocoder. Depending on the vocoder you are using, this might involve converting the audio files into WAV, PCM, or another specific format. Use an audio processing library like FFmpeg, SoX, or custom scripts to batch convert the audio files. 
3. Implement a step in your script where the converted audio files are passed through the vocoder. You can use a library or an external tool that applies the vocoder effect to these audio files automatically.
4. Once the files are processed, they need to be packed back into the game's format. You can automate the repackaging of these files back into the original or modded game archives, ensuring that the modified audio is correctly integrated.
5. Combine all these steps into a single automated script that handles extraction, conversion, vocoder processing, and repacking. This script can be run whenever you need to update or modify the in-game audio files.
